vortex-expr/src/transform/partition.rs

```diff
@@ -178,6 +181,10 @@ impl<'a> StructFieldExpressionSplitter<'a> {
             })
             .collect();
 
+        assert!(expression_accesses <= Some(partitions.len()));
+        // Ensure that there are as many partitions as there are accesses/fields in the scope
+        debug_assert_eq!(expression_accesses, Some(partitions.len()));
```

Comment:
these two assertions are the same? so on regular builds, we'll check once, on debug we'll check twice

Reply:
I don’t follow. The debug assertion is a tighter constraint checking perf. Where as the other checks correctness